sdk/communication/Azure.Communication.CallingServer/src/Models/Events/CallRecordingStateChanged.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Azure.Communication.CallingServer.Converters;
+using Azure.Core;
+
+namespace Azure.Communication.CallingServer
+{
+    /// <summary>
+    /// The call connected event.
+    /// </summary>
+    [CodeGenModel("RecordingStateChangedEvent", Usage = new string[] { "output" }, Formats = new string[] { "json" })]
+    public partial class CallRecordingStateChanged : CallAutomationEventBase
+    {
+        /// <summary>
+        /// THe recording state
+        /// </summary>
+        [JsonConverter(typeof(RecordingStateJsonConverter))]
+        public RecordingState State { get; set; }
```

Comment:
@minwoolee-msft this is the approach I took for the enum problem: I removed the nullable, plus had to add a JsonConverter to generate the right output.

Reply:
Just wondering, would the built-in [JsonStringEnumConverter](https://docs.microsoft.com/en-us/dotnet/api/system.text.json.serialization.jsonstringenumconverter?view=net-6.0) have worked just as well in this case?

Reply:
It didn't work on a test I did, with this error:
```
The converter specified on 'Azure.Communication.CallingServer.CallRecordingStateChanged.State' is not compatible with the type 'Azure.Communication.CallingServer.RecordingState'.
```

Maybe because these are `IEquatable` instead of basic enums.

Reply:
@cochi2 
Btw, you could probably take a look at the implementation of the JsonStringEnumConverter and use it to make a Generic version for IEquatable classes (in case there are more Enums that will need converters):

        /// <inheritdoc />
        public override JsonConverter CreateConverter(Type typeToConvert, JsonSerializerOptions options)
        {
            JsonConverter converter = (JsonConverter)Activator.CreateInstance(
                typeof(EnumConverter<>).MakeGenericType(typeToConvert),
                BindingFlags.Instance | BindingFlags.Public,
                binder: null,
                new object?[] { _converterOptions, _namingPolicy, options },
                culture: null)!;

            return converter;
        }

internal class EnumConverter<T> : JsonConverter<T>
        where T : struct, **Enum**
        
        Probably add IEquatable to T here